src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/volume/v4_9/CloudStackVolumePluginTest.java

```diff
@@ -72,22 +71,148 @@
+                .doDeleteInstance(Mockito.argThat(matcher), Mockito.eq(this.cloudStackUser));
+    }
+
+    // test case: When calling the deleteInstance method and occurs an FogbowException,
+    // it must verify if It returns a FogbowException.
+    @Test(expected = FogbowException.class)
+    public void testDeleteInstanceFail() throws FogbowException {
+        // set up
+        String instanceId = "instanceId";
+        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
+        Mockito.when(volumeOrder.getInstanceId()).thenReturn(instanceId);
+
+        Mockito.doThrow(new FogbowException())
+                .when(this.plugin).doDeleteInstance(Mockito.any(), Mockito.eq(this.cloudStackUser));
+
+        // exercise
+        this.plugin.deleteInstance(volumeOrder, this.cloudStackUser);
+    }
+
+    // test case: When calling the doDeleteInstance method with secondary methods mocked,
+    // it must verify if It reachs the right method without problems.
+    @Test
+    public void testDoDeleteInstanceSuccessfully() throws FogbowException, HttpResponseException {
+        // set up
+        DeleteVolumeRequest request = new DeleteVolumeRequest.Builder().build("");
+
+        String responseStr = "anything";
+        PowerMockito.mockStatic(CloudStackCloudUtils.class);
+        PowerMockito.when(CloudStackCloudUtils.doRequest(Mockito.eq(this.client),
+                Mockito.eq(request.getUriBuilder().toString()), Mockito.eq(this.cloudStackUser))).
+                thenReturn(responseStr);
+
+        boolean isSuccess = true;
+        DeleteVolumeResponse response = Mockito.mock(DeleteVolumeResponse.class);
+        Mockito.when(response.isSuccess()).thenReturn(isSuccess);
+        PowerMockito.mockStatic(DeleteVolumeResponse.class);
+        PowerMockito.when(DeleteVolumeResponse.fromJson(Mockito.eq(responseStr))).thenReturn(response);
+
+        // exercise
+        this.plugin.doDeleteInstance(request, this.cloudStackUser);
+
+        // verify
+        Mockito.verify(response, Mockito.times(TestUtils.RUN_ONCE)).isSuccess();
     }
 
+    // test case: When calling the doDeleteInstance method with secondary methods mocked and receive
+    // a response successless by the cloud, it must verify if It throws an UnexpectedException.
+    @Test
+    public void testDoDeleteInstanceFailWhenReturnSuccessless() throws FogbowException, HttpResponseException {
+        // set up
+        DeleteVolumeRequest request = new DeleteVolumeRequest.Builder().build("");
+
+        String responseStr = "anything";
+        PowerMockito.mockStatic(CloudStackCloudUtils.class);
+        PowerMockito.when(CloudStackCloudUtils.doRequest(Mockito.eq(this.client),
+                Mockito.eq(request.getUriBuilder().toString()), Mockito.eq(this.cloudStackUser))).
+                thenReturn(responseStr);
+
+        boolean isSuccessless = false;
+        String displayTextSuccessless = "successless";
```

Comment:
You can add a new constant to the **TestUtils** class to represent this string.

Reply:
In this case, I think that It's not necessary. This string does not have any value meaningful.